stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/RabbitMQTransactionContext.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.messaging.rabbitmq;
+
+import com.rabbitmq.client.Channel;
+import org.ballerinalang.bre.Context;
+import org.ballerinalang.model.values.BMap;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.util.transactions.BallerinaTransactionContext;
+import org.ballerinalang.util.transactions.TransactionLocalContext;
+import org.ballerinalang.util.transactions.TransactionResourceManager;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import javax.transaction.xa.XAResource;
+
+/**
+ * Channel wrapper class to handle Ballerina transaction logic.
+ *
+ * @since 0.995
+ */
+public class RabbitMQTransactionContext implements BallerinaTransactionContext {
+    private Channel channel;
+    private final String connectorId;
+    private long deliveryTag = -1;
+    private boolean autoAck = true;
+
+
+    /**
+     * Sets the value of autoAck.
+     *
+     * @param autoAck Ack mode of the consumer.
+     */
+    public void setAutoAck(boolean autoAck) {
+        this.autoAck = autoAck;
+    }
+
+    /**
+     * Sets the delivery tag of the message.
+     *
+     * @param deliveryTag Delivery tag of the message.
+     */
+    public void setDeliveryTag(long deliveryTag) {
+        this.deliveryTag = deliveryTag;
+    }
+
+    /**
+     * Initializes the rabbitmq transaction context.
+     *
+     * @param channelObject RabbitMQ Channel object.
+     * @param connectorId   Connector Id.
+     */
+    public RabbitMQTransactionContext(BMap<String, BValue> channelObject, String connectorId) {
+        channel = (Channel) channelObject.getNativeData(RabbitMQConstants.CHANNEL_NATIVE_OBJECT);
+        this.connectorId = connectorId;
+    }
+
+    @Override
+    public void commit() {
+        try {
+            if (deliveryTag != -1 && !autoAck) {
```

Comment:
Shall we use a constant to represent the meaning of this numerical value? Shall we check in all places?